src/app/SteamPatchesHook.ts

```diff
@@ -0,0 +1,82 @@
+//import logger from "../utils";
+import { SteamHook } from "./middleware";
+import { Clock, EventBus } from "./system";
+
+export { SteamPatchesHook }
+
+class SteamPatchesHook implements SteamHook {
+
+    private clock: Clock;
+    private eventBus: EventBus;
+
+    constructor(eventBus: EventBus, clock: Clock) {
+        this.eventBus = eventBus;
+        this.clock = clock;
+    }
+
+    public init() {
+        this.ReplaceAppInfoStoreOnAppOverviewChange();
+        this.ReplaceAppStoreMapAppsSet();
+    }
+
+    public unregister() {
+        this.RestoreOnAppOverviewChange();
+        this.RestoreAppStoreMapAppsSet();
+    }
+
+    // here we patch AppInfoStore OnAppOverviewChange method so we can prepare changed app overviews for the next part of the patch (AppOverview.InitFromProto)
+    private ReplaceAppInfoStoreOnAppOverviewChange() {
+        this.RestoreOnAppOverviewChange();
+        if (appInfoStore && !appInfoStore.OriginalOnAppOverviewChange) {
+            appInfoStore.OriginalOnAppOverviewChange = appInfoStore.OnAppOverviewChange;
+            let instance = this;
+            appInfoStore.OnAppOverviewChange = function (apps: Array<any>) {
+                //logger.trace(`AppInfoStore.OnAppOverviewChange (${apps ? "[]" : "null"})`);
+                let appIds = apps.filter((_: any) => typeof _.appid() === "number").map((_: any) => _.appid() as number);
+                instance.eventBus.emit({
+                    type: "AppInfoStore.OnAppOverviewChange",
+                    createdAt: instance.clock.getTimeMs(),
+                    appIds: appIds,
+                });
+                //logger.trace(`AppInfoStore.OnAppOverviewChange: calling original`);
+                this.OriginalOnAppOverviewChange(apps);
+            };
+        }
+    }
+
+    private RestoreOnAppOverviewChange() {
+        if (appInfoStore && appInfoStore.OriginalOnAppOverviewChange) {
+            appInfoStore.OnAppOverviewChange = appInfoStore.OriginalOnAppOverviewChange;
+            appInfoStore.OriginalOnAppOverviewChange = null;
+        }
+    }
+
+    // here we patch AppStore m_mapApps Map set method so we can ovewrite playtime before setting AppOverview
+    private ReplaceAppStoreMapAppsSet() {
+        this.RestoreAppStoreMapAppsSet();
+        if (appStore.m_mapApps && !appStore.m_mapApps.originalSet) {
+            appStore.m_mapApps.originalSet = appStore.m_mapApps.set;
+            let instance = this;
+            let appStoreInstance = appStore;
+            appStore.m_mapApps.set = function (appId: number, appOverview: any) {
```

Comment:
Very [similar](https://github.com/FrogTheFrog/moondeck/blob/main/src/lib/appdetailspatcher.ts#L228) approach used in `moondeck`, do you tested that both plugins are working good with each other?

Reply:
Hm, don't use the plugin but in principle it should be fine. The patching is done differently. I might check it out.